### PR TITLE
Reactions bridging

### PR DESCRIPTION
**Pull request by Cadair**
~builds on #130 which builds on #129~

TODO:

- [ ] Need to use aggregations API to handle slack -> matrix reaction removal.

### CODE REVIEW

**Comment by Cadair** [comment]
I am not sure I can quite get the aggregations stuff in synapse to play ball for this, but I am trying.

**Comment by Cadair** [comment]
it seems matrix -> slack reaction removal might be broken.

**Comment by Cadair** [comment]
So it seems that slack has alternative names for a bunch of emoji i.e `:+1:` and `:thumbsup:` and this works with `:+1:` but not `:thumbsup:`. 


AHGHGGGGHHHHHHHLLLLLL

**Comment by Cadair** [comment]
Fixed.

**Comment by Cadair** [comment]
this is now based on #153

File: lib/BridgedRoom.js [hunk 1]
```diff
@@ -190,6 +191,58 @@ BridgedRoom.prototype.toEntry = function() {
     return entry;
 };
 
+BridgedRoom.prototype.onMatrixReaction = async function(message) {
+    if (!this._slack_bot_token) return Promise.resolve();
```
Comment (Half-Shot) [inline lib/BridgedRoom.js]: This is an async function, just return

File: lib/BridgedRoom.js [hunk 2]
```diff
@@ -389,6 +446,68 @@ BridgedRoom.prototype.onSlackMessage = async function(message) {
...
+};
+
+BridgedRoom.prototype.getMatrixReactionsBySenderByKey = async function(eventId, sender, key) {
+    // The objective of this function is to hit the new
+    // rooms/roomId/aggregations endpoint for all the m.reaction events with the
+    // specified key.
+    // Then we walk that list and find the one with the sender we care about, and return that event.
+
+    // TODO: Not sure this works, having trouble testing it against a sufficiently new synapse.
+    return;
```
Comment (Half-Shot) [inline lib/BridgedRoom.js]: Please just remove the func if it doesn't work. I'm not sure it's worth bloating the code base out with future things if the spec isn't even merged yet.

File: lib/SlackEventHandler.js [hunk 3]
```diff
@@ -146,7 +152,7 @@ SlackEventHandler.prototype.doChannelUserReplacements = async function(msg, text
  * @param {string} params.event.ts The unique (per-channel) timestamp
  */
 SlackEventHandler.prototype.handleMessageEvent = async function (params) {
-    var room = this._main.getRoomBySlackChannelId(params.event.channel);
+    var room = this._main.getRoomBySlackChannelId(params.event.channel || params.event.item.channel);
```
Comment (Half-Shot) [inline lib/SlackEventHandler.js]: VAR

**Comment by Half-Shot** [comment]
FYI, we decided to not do redactions of reactions slack -> matrix because it involves having to listen for your own reactions over the events API and it can wait for a future PR.

**Pull request merged**